cli/flox-rust-sdk/src/providers/flox_cpp_utils.rs

```diff
@@ -0,0 +1,261 @@
+            .lock()
+            .unwrap()
+            .pop_front()
+            .expect("no more mock results");
+
+        debug!(
+            system=system.as_ref(),
+            installable=installable.as_ref(),
+            mocked_result=?mocked_result,
+            "responding with mocked result"
+        );
+
+        mocked_result
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use std::path::Path;
+
+    use super::*;
+
+    /// Returns the path to a bundled flake that contains a number of test packages
+    /// for sped up evaluation
+    fn local_test_flake() -> String {
+        let manifest_root = Path::new(env!("CARGO_MANIFEST_DIR"));
+        let local_test_flake_path = manifest_root
+            .join("../../pkgdb/tests/data/lock-flake-installable")
+            .canonicalize()
+            .unwrap();
+        local_test_flake_path.to_str().unwrap().to_string()
+    }
+
+    /// Test that the output of `pkgdb lock-flake-installable` can be deserialized
+    /// into a [LockedFlakeInstallble] struct.
+    #[test]
+    fn test_output_format() {
+        // `$system` is set by the nix devshell
+        let system = env!("system");
+        let installable = format!("{flake}#hello", flake = local_test_flake());
+
+        // make sure the deserialization is not accidentally optimized away
+        let _locked = Pkgdb
+            .lock_flake_installable(system, installable)
+            .expect("locking local test flake should succeed");
+    }
+
+    // Tests against locking errors thown by pkgdb.
+    //
+    // There is currently no coverage of error cases in the pkgdb unit tests,
+    // because it's not yet clear how detailed we want tests to be.
+    // Currently, flake lock errors are caught and thrown as `LockFlakeInstallableException`,
+    // while most evaluation errors are thrown as plain nix errors.
+    // While we should have coverage of error cases in pkgdb as well,
+    // we need tests on the rust side that ensure
+    // that the errors are mapped to the right [FlakeInstallableError] variant.
+    // These also tests the error handling in the pkgdb implementation, indirectly.
+    //
+    // region: pkgdb errors
+
+    #[test]
+    fn test_catches_absent_flake() {
+        let system = env!("system");
+        let installable = "github:flox/trust-this-wont-be-added#hello";
+
+        let result = Pkgdb.lock_flake_installable(system, installable);
+        assert!(
+            matches!(result, Err(FlakeInstallableError::LockInstallable(_))),
+            "{result:#?}"
+        );
+    }
+
+    #[test]
+    fn test_catches_absent_flake_output() {
+        let system = env!("system");
+        let installable = format!("{flake}#nonexistent", flake = local_test_flake());
+
+        let result = Pkgdb.lock_flake_installable(system, installable);
+
+        assert!(
+            matches!(result, Err(FlakeInstallableError::LockInstallable(_))),
+            "{result:#?}"
+        );
+    }
+
+    // endregion: pkgdb errors
```

Comment:
```suggestion
```

Reply:
the region does start earlier up ([end]region allows folding in vscode, that may be a non reason if we want to remove these)

Reply:
No that's fine